### PR TITLE
Review order of RCPT-ACLs

### PR DESCRIPTION
**Pull request by Udera**
Changes:

* accept authenticated senders before checking blacklists
* put everything optional into `vexim-acl-check-rcpt`

### CODE REVIEW

File: docs/configure [hunk 1]
```diff
@@ -363,22 +360,17 @@ acl_check_rcpt:
...
-  # deny    message       = rejected because $sender_host_address is in a black list at $dnslist_domain\n$dnslist_text
-  #         dnslists      = black.list.example
-  #
-  # warn    message       = X-Warning: $sender_host_address is in a black list at $dnslist_domain
-  #         log_message   = found in $dnslist_domain
-  #         dnslists      = black.list.example
-  #############################################################################
+
+  # Accept if the message arrived over an authenticated connection, from
+  # any host. Again, these messages are usually from MUAs, so recipient
```
Comment (rimas-kudelis) [inline docs/configure]: What's the "Again" for?

File: docs/configure [hunk 2]
```diff
@@ -407,11 +400,6 @@ acl_check_rcpt:
 
   accept  hosts         = +relay_from_hosts
```
Comment (rimas-kudelis) [inline docs/configure]: Oh, I see what the "Again" is for. Perhaps the block above should also be moved upwards?

**Comment by Udera** [comment]
Moved the relay-host block upwards. Only thing I'm not sure about, how does exim get the hostname? If it is just taken from the `EHLO` during the SMTP session, this value is not trustworthy. You could guess relay hosts. We should verify this to avoid someone becoming an open relay.

File: docs/configure [hunk 3]
```diff
@@ -363,22 +360,22 @@ acl_check_rcpt:
...
+
+  accept  authenticated = *
+
+  # Accept if the message comes from one of the hosts for which we are an
+  # outgoing relay. Recipient verification is omitted here, because in many
+  # cases the clients are dumb MUAs that don't cope well with SMTP error
+  # responses. If you are actually relaying out from MTAs, you should probably
+  # add recipient verification here.
+
+  accept  hosts         = +relay_from_hosts
```
Comment (rimas-kudelis) [inline docs/configure]: Just fix this whitespace, and I think this can be merged in.

**Comment by rimas-kudelis** [comment]
DNS, I would guess?

File: docs/configure (hunk 3 above)
Comment (Udera) [inline docs/configure]: which one?

**Comment by Udera** [comment]
Yes, perhaps. A reverse DNS lookup on the IP address. Or query the DNS name and check if one of the IPs is identical with the sender. Perhaps exim is already handling this, but we should check this.

File: docs/configure (hunk 3 above)
Comment (rimas-kudelis) [inline docs/configure]: The huge one before the `=`

Comment (rimas-kudelis) [inline docs/configure]: Also, the comments are written in a way where `accept authenticated` goes after `accept hosts`. If you want these conditions in this order, you should update the comments. Although I'd just switch them back to their original order, I don't think it's goin' to make much difference.

Comment (Udera) [inline docs/configure]: > The huge one before the =

They are all aligned at the same position. Have a look at the following rules, there it is the same.

Comment (rimas-kudelis) [inline docs/configure]: okay, but the other nit still applies

**Pull request merged**